app/models/katello/concerns/hostgroup_extensions.rb

```diff
@@ -88,14 +88,19 @@ def content_and_puppet_match?
       end
 
       def equivalent_kickstart_repository
-        return unless operatingsystem && kickstart_repository
+        return unless operatingsystem &&
+                      kickstart_repository &&
+                      operatingsystem.class.method_defined?(:kickstart_repos)
```

Comment:
Later on we are calling `kickstart_repos` on an instance of an OperatingSystem - what made you check the class for method defined?

Reply:
I see it's because of the way method_defined? works!